### DISCUSSION TITLE
Change project window size property from two `int`s to one `Vector2i`

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#4349] [Editor]
For `3.x`, two ints make sense because we only have `Vector2`.

For `4.0`, I think they can be changed to use one `Vector2i`. So it'll display the edit control for `Vector2i` and benefit from features like Vector X-Y value linking.

**Comment by Calinou** [comment]
One downside is that the inspector does not support min/max range hints for Vector2 yet, so the setter must implement clamping using the `CLAMP()` macro. It's not a huge downside, just something to keep in mind.

**Reply by Mickeon** [reply]
I believe, now that exporting properties has been streamlined with annotations, it's time to begin proposing more export options for property types that aren't Integers, Floats or Strings... so that manipulating their setters for very simple operations may not be required nearly as much.
